### PR TITLE
misc

### PR DESCRIPTION
**Pull request by pierrec**
Changes to conv::char32_to_utf8_unsafe(); add tests for DString
add tests for DString

### CODE REVIEW

File: lib/std/core/dstring.c3 [hunk 1]
```diff
@@ -308,18 +308,18 @@ fn void DString.insert_at(&self, usz index, String s)
 	data.len += s.len;
 
 	char* start = data.chars[index:s.len].ptr; // area to insert into
-	mem::move(start + s.len, start, len - index); // move existing data
+	mem::move((void*)start + s.len, start, len - index); // move existing data
```
Comment (lerno) [inline lib/std/core/dstring.c3:311]: Why the void casts?

Comment (pierrec) [reply lib/std/core/dstring.c3:310]: When working on DString with the header at the tail, I had miscalculations if I didnt cast.

Comment (lerno) [reply lib/std/core/dstring.c3:310]: That... should never be. Do you have a repro? `void*` and `char*` works the same way.

Comment (pierrec) [reply lib/std/core/dstring.c3:310]: Wait. I think it was a StringData*... I'll remove the void*.

**Comment by lerno** [comment]
Thank you.

**Pull request merged**